ApptentiveFeedback/ApptentiveFeedback/Additions/ApptentiveCallbacks.cs

```diff
@@ -0,0 +1,350 @@
+        {
+            ApptentiveKit.Android.Apptentive.RemoveCustomPersonData(key);
+        }
+
+        /// <summary>
+        /// Add a custom data String to the Person. Custom data will be sent to the server, is displayed
+        /// in the Conversation view, and can be used in Interaction targeting.
+        /// </summary>
+        /// <param name="key">The key to store the data under.</param>
+        /// <param name="value">A String value.</param>
+        public static void AddCustomDeviceData(String key, String value)
+        {
+            ApptentiveKit.Android.Apptentive.AddCustomDeviceData(key, value);
+        }
+
+        /// <summary>
+        /// Add a custom data int to the Person. Custom data will be sent to the server, is displayed
+        /// in the Conversation view, and can be used in Interaction targeting.
+        /// </summary>
+        /// <param name="key">The key to store the data under.</param>
+        /// <param name="value">A int value.</param>
+        public static void AddCustomDeviceData(String key, int value)
+        {
+            Java.Lang.Integer intValue = Java.Lang.Integer.ValueOf(value);
+            Java.Lang.Number numberValue = (Java.Lang.Number)intValue;
+
+            ApptentiveKit.Android.Apptentive.AddCustomDeviceData(key, numberValue);
+        }
+
+        /// <summary>
+        /// Add a custom data bool to the Person. Custom data will be sent to the server, is displayed
+        /// in the Conversation view, and can be used in Interaction targeting.
+        /// </summary>
+        /// <param name="key"></param>
+        /// <param name="value"></param>
+        public static void AddCustomDeviceData(String key, bool value)
+        {
+            Java.Lang.Boolean javaValue = Java.Lang.Boolean.ValueOf(value);
+            ApptentiveKit.Android.Apptentive.AddCustomDeviceData(key, javaValue);
+        }
+
+        /// <summary>
+        /// Sends a text message to the server. This message will be visible in the conversation view
+        /// on the server, but will not be shown in the client's Message Center.
+        /// </summary>
+        /// <param name="text">The message you wish to send.</param>
+        public static void SendAttachmentText(String text)
+        {
+            ApptentiveKit.Android.Apptentive.SendAttachmentText(text);
+        }
+
+        /// <summary>
+        /// Sends a file to the server. This file will be visible in the conversation view on the server,
+        /// but will not be shown in the client's Message Center. A local copy of this file will be made
+        /// until the message is transmitted, at which point the temporary file will be deleted.
+        /// </summary>
+        /// <param name="uri">The URI path of the local resource file.</param>
+        public static void SendAttachmentFile(String uri)
+        {
+            ApptentiveKit.Android.Apptentive.SendAttachmentFile(uri);
+        }
```

Comment:
Curious if this was tested. It's not implemented in Cordova.

Reply:
Still curious if this was tested?

Reply:
Tested and will be added in XoloFacts